src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -2030,9 +2018,25 @@ Interpolate::Interpolate(const std::shared_ptr<ov::Node>& op, const GraphContext
                 axes = ov::as_type_ptr<const ov::op::v0::Constant>(interp->get_input_node_shared_ptr(AXES_ID))
                            ->cast_vector<int>();
             } else {
-                axes.resize(dataRank);
-                for (int i = 0; i < static_cast<int>(dataRank); i++) {
-                    axes[i] = i;
+                auto outputShape = getOutputShapeAtPort(0);
+                auto avoidReorder = [](const auto& inputDataShape, const auto& outputShape) {
+                    auto lastInDim = inputDataShape.getDims().back();
+
+                    // Dynamic shape
+                    if (lastInDim == Shape::UNDEFINED_DIM) {
+                        return false;
+                    }
+                    auto lastOutDim = outputShape.getDims().back();
+                    return static_cast<bool>(lastOutDim == lastInDim);
+                };
+                if (dataRank == 4 && avoidReorder(inputDataShape, outputShape)) {
+                    interpAttrs.NCHWAsNHWC = true;
```

Comment:
1. This only support static shape case, why not set this flag in `isAxesSpecified` case as well, where can mark this flag same as pillow_mode(axis is provided and is [1,2] in 4d case). In that case, we can do this optimization on dynamic shape as well.
2. We do not support 5d with axis[1,2,3] in this pr, right?
3. Please add comment why name `avoidReorder` is used.